agent/config/runtime.go

```diff
@@ -48,160 +55,1173 @@ type RuntimeConfig struct {
-	RPCRateLimit                rate.Limit
-	RaftProtocol                int
-	ReconnectTimeoutLAN         time.Duration
-	ReconnectTimeoutWAN         time.Duration
-	RejoinAfterLeave            bool
-	RetryJoinIntervalLAN        time.Duration
-	RetryJoinIntervalWAN        time.Duration
-	RetryJoinLAN                []string
-	RetryJoinMaxAttemptsLAN     int
-	RetryJoinMaxAttemptsWAN     int
-	RetryJoinWAN                []string
-	SegmentName                 string
-	Segments                    []structs.NetworkSegment
-	SerfAdvertiseAddrLAN        *net.TCPAddr
-	SerfAdvertiseAddrWAN        *net.TCPAddr
-	SerfBindAddrLAN             *net.TCPAddr
-	SerfBindAddrWAN             *net.TCPAddr
-	SerfPortLAN                 int
-	SerfPortWAN                 int
-	ServerMode                  bool
-	ServerName                  string
-	ServerPort                  int
-	Services                    []*structs.ServiceDefinition
-	SessionTTLMin               time.Duration
-	SkipLeaveOnInt              bool
-	StartJoinAddrsLAN           []string
-	StartJoinAddrsWAN           []string
-	SyslogFacility              string
-	TLSCipherSuites             []uint16
-	TLSMinVersion               string
+
+	// Node name is the name we use to advertise. Defaults to hostname.
+	//
+	// NodeName is exposed via /v1/agent/self from here and
+	// used in lots of places like CLI commands. Treat this as an interface
+	// that must be stable.
+	// todo(fs): do we need this warning? RuntimeConfig was meant to be refactorable.
+	//
+	// hcl: node_name = string
+	// flag: -node string
+	NodeName string
+
+	// AdvertiseAddrLAN is the address we use for advertising our Serf, and
+	// Consul RPC IP. The address can be specified as an ip address or as a
+	// go-sockaddr template which resolves to a single ip address. If not
+	// specified, the bind address is used.
+	//
+	// hcl: advertise_addr = string
+	AdvertiseAddrLAN *net.IPAddr
+
+	// AdvertiseAddrWAN is the address we use for advertising our Serf, and
+	// Consul RPC IP. The address can be specified as an ip address or as a
+	// go-sockaddr template which resolves to a single ip address. If not
+	// specified, the bind address is used.
+	//
+	// hcl: advertise_addr_wan = string
+	AdvertiseAddrWAN *net.IPAddr
+
+	// BindAddr is used to control the address we bind to.
+	// If not specified, the first private IP we find is used.
+	// This controls the address we use for cluster facing
+	// services (Gossip, Server RPC)
+	//
+	// The value can be either an ip address or a go-sockaddr
+	// template which resolves to a single ip address.
+	//
+	// hcl: bind_addr = string
+	// flag: -bind string
+	BindAddr *net.IPAddr
+
+	// Bootstrap is used to bring up the first Consul server, and
+	// permits that node to elect itself leader
+	//
+	// hcl: bootstrap = (true|false)
+	// flag: -bootstrap
+	Bootstrap bool
+
+	// BootstrapExpect tries to automatically bootstrap the Consul cluster,
+	// by withholding peers until enough servers join.
```

Comment:
"by having servers wait to bootstrap until enough servers join, and then performing the bootstrap process automatically. They will disable their automatic bootstrap process if they detect any servers that are part of an existing cluster, so it's safe to leave this set to a non-zero value."